### PR TITLE
[Dashboard] Replace SiGithub

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on replacing the usage of the `SiGithub` icon from `react-icons` with a custom `GithubIcon` component across multiple files for consistency and improved maintainability.

### Detailed summary
- Replaced `SiGithub` with `GithubIcon` in:
  - `GameCard.tsx`
  - `app-footer.tsx`
  - `edit-profile.tsx`
  - `socialLinks.tsx`
  - `PublisherSocials.tsx`
- Added `GithubIcon` component in `GithubIcon.tsx` with SVG definition.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4811?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4811** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4811?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4810** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/components/icons/brand-icons/GithubIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const Github = (props: SVGProps<SVGSVGElement>) => {
+  return (
+    <svg
+      role="img"
+      viewBox="0 0 24 24"
+      xmlns="http://www.w3.org/2000/svg"
+      fill="currentColor"
+      width="24px"
+      height="24px"
+      {...props}
```
Comment (MananTank) [inline apps/dashboard/src/components/icons/brand-icons/GithubIcon.tsx:12]: remove px from width and height - just 24

**Review by MananTank — CHANGES_REQUESTED** [review]
Lets stick to tailwind classes for setting the size. `size-4`, `size-5` etc

File: apps/dashboard/src/components/contract-components/publisher/edit-profile.tsx [hunk 2]
```diff
@@ -214,7 +214,7 @@ export const EditProfile: React.FC<EditProfileProps> = ({
           <FormControl isInvalid={!!errors.github}>
             <FormLabel>
               <div className="flex flex-row gap-2">
-                <Icon as={SiGithub} boxSize={4} />
+                <Github width={16} height={16} />
```
Comment (MananTank) [inline apps/dashboard/src/components/contract-components/publisher/edit-profile.tsx:217]: Lets stick to tailwind classes for setting the size

File: apps/dashboard/src/components/product-pages/common/GameCard.tsx [hunk 3]
```diff
@@ -96,7 +96,7 @@ export const GameCard: React.FC<GameCardProps> = ({
             }}
             size="sm"
             variant="ghost"
-            icon={<SiGithub />}
+            icon={<Github width={16} height={16} />}
```
Comment (MananTank) [inline apps/dashboard/src/components/product-pages/common/GameCard.tsx:99]: use size-4 class instead

File: apps/dashboard/src/components/footer/socialLinks.tsx [hunk 4]
```diff
@@ -55,6 +55,6 @@ export const SOCIALS: socialLinkInfo[] = [
     link: "https://github.com/thirdweb-dev",
     ariaLabel: "GitHub",
     label: "gitHub",
-    icon: <SiGithub fontSize={socialIconSize} />,
+    icon: <Github fontSize={socialIconSize} width={20} height={20} />,
```
Comment (MananTank) [inline apps/dashboard/src/components/footer/socialLinks.tsx:58]: use size-5 class instead

File: apps/dashboard/src/components/contract-components/publisher/PublisherSocials.tsx [hunk 5]
```diff
@@ -67,7 +67,7 @@ export const PublisherSocials: React.FC<PublisherSocialsProps> = ({
         }
         bg="transparent"
         aria-label="github"
-        icon={<Icon as={SiGithub} />}
+        icon={<Github width={16} height={16} />}
```
Comment (MananTank) [inline apps/dashboard/src/components/contract-components/publisher/PublisherSocials.tsx:70]: use size-4 class instead

File: apps/dashboard/src/@/components/blocks/app-footer.tsx [hunk 6]
```diff
@@ -35,7 +35,7 @@ export function AppFooter(props: AppFooterProps) {
           <div className="flex flex-row gap-3">
             <Button size="icon" variant="ghost" asChild className="size-9">
               <Link href="https://github.com/thirdweb-dev" target="_blank">
-                <SiGithub className="size-5 text-muted-foreground" />
+                <Github className="size-5 fill-muted-foreground" />
```
Comment (MananTank) [inline apps/dashboard/src/@/components/blocks/app-footer.tsx:38]: prefer `text-` unless `fill` is absolute necessary

File: apps/dashboard/src/components/icons/brand-icons/GithubIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import type { SVGProps } from "react";
+
+export const Github = (props: SVGProps<SVGSVGElement>) => {
```
Comment (MananTank) [inline apps/dashboard/src/components/icons/brand-icons/GithubIcon.tsx:3]: name the component `GithubIcon` instead

**Review by MananTank — APPROVED** [review]
thanks!

**Pull request merged**